mocktail/src/server.rs

```diff
@@ -58,11 +60,17 @@ impl MockServer {
         self
     }
 
+    /// Sets the server configuration.
+    pub fn with_config(mut self, config: MockServerConfig) -> Self {
+        self.config = config;
+        self
+    }
+
     pub async fn start(&self) -> Result<(), Error> {
         if self.addr().is_some() {
             return Err(Error::ServerError("already running".into()));
         }
-        let port = find_available_port().unwrap();
+        let port = find_available_port(&self.config).unwrap();
         let addr = SocketAddr::from(([0, 0, 0, 0], port));
         let base_url = Url::parse(&format!("http://{}", &addr)).unwrap();
         info!("starting {} [{}] server on {addr}", self.name(), &self.kind);
```

Comment:
Thinking about this more, let's drop the `find_available_port()` helper and just try ports here. Currently, we end up binding twice, and the second attempt here could _potentially_ fail. This also returns an error in case it fails.

```suggestion
        let config = &self.config;

        let mut counter = 0;
        let mut rng = rand::rng();
        let listener = loop {
            let port: u16 = rng.random_range(config.port_range_start..config.port_range_end);
            let addr = SocketAddr::from((config.listen_addr, port));
            if let Ok(listener) = TcpListener::bind(&addr).await {
                break listener;
            }
            if counter == config.bind_max_retries {
                return Err(Error::ServerError("failed to bind to port".into()));
            }
            counter += 1;
        };

        let addr = listener.local_addr()?;
        info!("started {} [{}] server on {addr}", self.name(), &self.kind);
        let base_url = Url::parse(&format!("http://{}", &addr)).unwrap();

```